packages/nimble-components/src/table-column/icon/tests/table-column-icon.mdx

```diff
@@ -17,7 +17,15 @@ The <Tag name={tableColumnIconTag}/> column renders string, number, or boolean v
 <Canvas of={tableColumnIconStories.iconColumn} />
 <Controls of={tableColumnIconStories.iconColumn} />
 
-## Blazor Usage
+## Usage
+
+### Best Practices
+
+-   Avoid mixing `undefined` and `null` as values for the same field. When grouping this will lead to two groups (one for `null` values and one for `undefined` values) that both have the text `"No value"`.
+-   Avoid using values that do not correspond to a mapping for the column.
+-   To display an empty cell but have a non-blank group row, create a mapping of the record value to an `undefined` icon.
```

Comment:
I'm not sure how mapping values to an `undefined` icon improves sorting behavior, but I included the rest of the suggestion.

Reply:
I thought that for both grouping and sorting, this helped by normalizing the value used to group/sort rather than a bunch of different values. But I might have gotten that wrong.

Reply:
You are right that since sorting is done by record value, having a bunch of different record values that display as the same thing (whether they display as the same mapped value or as an empty unmapped value) would be weird when it comes to sorting and grouping. However, I don't think having mappings for everything solves that problem. I'll add an additional bullet point here that says something like _Avoid having multiple values that map to the same label because grouping and sorting is done on the record value rather than the mapped label._